test/regExpTest.coffee

```diff
@@ -0,0 +1,71 @@
+expect = require('chai').expect
+
+describe 'regExp matches URI', ->
+  regExp = new RegExp("^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?", "i")
```

Comment:
I've decided to use: https://github.com/kevva/url-regex

It seems to work well enough.